cpg_infra/config/config.py

```diff
@@ -79,16 +79,19 @@ class Hail(DeserializableDataclass):
         @dataclasses.dataclass(frozen=True)
         class GCP(DeserializableDataclass):
             hail_batch_url: str
+            hail_auth_url: str
             git_credentials_secret_name: str | None = None
             git_credentials_secret_project: str | None = None
             wheel_bucket_name: str | None = None
 
         @dataclasses.dataclass(frozen=True)
         class Azure(DeserializableDataclass):
             hail_batch_url: str
+            hail_auth_url: str
 
         gcp: GCP
         azure: Azure | None = None
+        username_prefix: str | None = ''
```

Comment:
Should this default to `None` or not allow None?

```suggestion
        username_prefix: str | None = None
```

Reply:
good pickup, I wasn't handling None's properly for the username prefix, have fixed that now